tests/resources/test_bills.py

```diff
@@ -59,6 +60,21 @@ def test_cancel_bill(client):
     assert updated_transaction.status == 'refunded'
 
 
+@pytest.mark.vcr
+def test_cancel_bill_fail(client):
+    transactions = client.transactions.list()
+    transactions = list(filter(lambda t: t.status == 'fulfilled',
+                               transactions))
+    if transactions:
```

Comment:
You're skipping the test entirely if this evaluates to `False`. That defeats the entire point of the test.

My recommendation is to make the test self-reliant:
1. create bill
1. pay bill (that has a delay to become fulfilled), which returns a transaction
1. cancel transaction
1. cancel transaction again

and ...
1. create bill
1. pay bill that is immediately fulfilled
1. cancel transaction

Both cases should throw `InvalidOperation` (inherited from `UnprocessableEntity`) with the message:
`f'Unable to cancel a transaction with the status: {transaction.status}'`